lib/ansible/executor/task_executor.py

```diff
@@ -321,6 +323,7 @@ def _run_loop(self, items):
         items_len = len(items)
         for item_index, item in enumerate(items):
             task_vars['ansible_loop_var'] = loop_var
+            task_vars['notify_scope'] = notify_scope
```

Comment:
do not pass as a variable in the task vars, we want to avoid mixing those as it creates all kinds of issues (now you just created a reserved var name that will override user's values)

Reply:
Fixed in the last commit.